test/socket.js

```diff
@@ -44,4 +44,19 @@ describe('socket', function(){
     sock.close();
   });
 
+  it('should throw a javascript error if it hits the system file descriptor limit', function() {
+    var i, socks = [], numSocks = 10000;
+    function hitlimit() {
+      for (i = 0; i < numSocks; i++) {
+        socks.push(zmq.socket('router'));
+      }
+    }
+    hitlimit.should['throw'];
+    for (i = 0; i < numSocks; i++) {
```

Comment:
I tested a console.log between lines 54 and 55 and it appeared in the test output. I'll make another commit to not stupidly iterate over 9000 undefined array entries.